chain/jsonrpc-primitives/src/types/transactions.rs

```diff
@@ -0,0 +1,132 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Debug, Clone)]
+pub struct RpcBroadcastTransactionRequest {
+    pub signed_transaction: near_primitives::transaction::SignedTransaction,
+}
+
+#[derive(Debug)]
+pub struct RpcTransactionStatusCommonRequest {
+    pub transaction_info: TransactionInfo,
+}
+
+#[derive(Clone, Debug)]
+pub enum TransactionInfo {
+    Transaction(near_primitives::transaction::SignedTransaction),
+    TransactionId {
+        hash: near_primitives::hash::CryptoHash,
+        account_id: near_primitives::types::AccountId,
+    },
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum RpcTransactionError {
+    #[error("An error happened during transaction execution: {context:?}")]
+    InvalidTransaction { context: near_primitives::errors::InvalidTxError },
+    #[error("Node doesn't track this shard. Cannot determine whether the transaction is valid")]
+    DoesNotTrackShard,
+    #[error("Transaction with hash {transaction_hash} was routed")]
+    RequestRouted { transaction_hash: String },
+    #[error("Transaction {requested_transaction_hash} doesn't exist")]
+    UnknownTransaction { requested_transaction_hash: near_primitives::hash::CryptoHash },
+    #[error("The node reached its limits. Try again later. More details: {debug_info}")]
+    InternalError { debug_info: String },
+    #[error("Timeout")]
+    TimeoutError,
+    // NOTE: Currently, the underlying errors are too broad, and while we tried to handle
+    // expected cases, we cannot statically guarantee that no other errors will be returned
+    // in the future.
+    // TODO #3851: Remove this variant once we can exhaustively match all the underlying errors
+    #[error("It is a bug if you receive this error type, please, report this incident: https://github.com/near/nearcore/issues/new/choose. Details: {error_message}")]
+    Unreachable { error_message: String },
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RpcTransactionResponse {
+    #[serde(flatten)]
+    pub final_execution_outcome: near_primitives::views::FinalExecutionOutcomeViewEnum,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RpcBroadcastTxSyncResponse {
+    pub transaction_hash: String,
+    pub is_routed: bool,
```

Comment:
I feel that `is_routed` is misleading, and it is better to just drop it and only return `transaction_hash` while the method is still EXPERIMENTAL.

Reply:
Do we still want to have a structure for that instead of returning `CryptoHash` on the jsonrpc side?

Reply:
@khorolets yes, we still want to have a structure since with structure we always have an opportunity to provide extra context if necessary in the future.